lib/elixir/lib/range.ex

```diff
@@ -40,7 +40,8 @@ defmodule Range do
   defstruct first: nil, last: nil
 
   @type t :: %Range{}
-  @type t(first, last) :: %Range{first: first, last: last}
+  @type t(first :: integer, last :: integer) :: %Range{first: integer, last: integer}
```

Comment:
I see. so this is my second shot

```elixir
  @type t :: %Range{first: integer, last: integer}
  @type t(first, last) :: %Range{first: first, last: last}
```

but I'm wondering if `t/2` can in some way say at least it has to be an integer, and allow users to specify a subset of it if needed.
wouldn't something like `@type t(first, last) :: %Range{first: first, last: last} when [first: integer, last: integer]` work?